### PR TITLE
Add Windows support

### PR DESCRIPTION
**Pull request by kustomzone**
MinGW has [links](http://mingw-w64.org/doku.php/download) for compiling on all the main platforms, so included that in the readme.

Found the VS Build Tools 2019 **also** has the SDK addon, so didn't need to explain about VS Community / Professional / Enterprise etc. Phew!

I'm guessing ["Linking to issues"](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) is done by the repo owner.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,7 +86,18 @@ From a peer-to-peer alternative to github and similar third party git portal ser
...
+* [NodeJS](https://nodejs.org/en/) (v12 or later)
+* [Rust](https://www.rust-lang.org/tools/install)
+* [Rustup](https://rustup.rs/)
+
+### Compiling on Windows and other Platforms
+As suggested on the Rust install page, a compiler such as the one included with Windows Visual Studio may be required.
+For Windows, Visual Studio Build Tools 2019 requires a Windows SDK for your platform via [Visual Studio Installer](https://visualstudio.microsoft.com/downloads/). Choose Modify, C++ Build Tools, then add the Windows SDK.
+A console compiler such as [MingW](http://mingw-w64.org/doku.php/download)'s GCC (GNU Compiler Collection) supports most other platforms.
+Alternative IDEs for writing Rust such as [Atom](https://atom.io/) which have [Rust support](https://atom.io/packages/ide-rust) are available.
+
```
Comment (happybeing) [inline README.md]: I like this, it's clear and instructive. I suggest we keep the common sections together, so move the block about other OS below the "WebAssembly (Wasm)" section, just above "Get the code".

File: package.json [hunk 2]
```diff
@@ -19,25 +19,27 @@
...
+    "cleanup": "cd public && rm -rf wasm && mkdir wasm && cd .. && mv *.wasm public/wasm/ && mv wasi-* src/ && yarn docs",
     "dev": "webpack-dev-server --watch-content-base --content-base public",
-    "dev-wasm": "cd src/wasi-rust/ && cargo build --target=wasm32-wasi && mkdir -p ../../public/wasm",
-    "dev-wasm-bindgen": "yarn dev-wasm && wasm-bindgen --no-typescript src/wasi-rust/target/wasm32-wasi/debug/wasi-gitportal.wasm --debug --keep-debug  --out-dir . && yarn wbg-to-wasi && mv *.wasm public/wasm/ && mv wasi-* src",
-    "watch-wasm-bindgen": "watchlist src/wasi-rust -i src/wasi-rust/target -- yarn dev-wasm-bindgen",
-    "wbg-to-wasi": "./scripts/wbg_to_wasi.js --path ./wasi-js/ wasi-gitportal_bg.js wasi-gitportal_bg_wasi.js"
+    "wasm-bindgen-dev": "wasm-bindgen --no-typescript src/wasi-rust/target/wasm32-wasi/debug/wasi-gitportal.wasm --debug --keep-debug --out-dir .",
+    "wasm-bindgen-release": "wasm-bindgen --no-typescript src/wasi-rust/target/wasm32-wasi/release/wasi-gitportal.wasm --browser --out-dir .",
+    "watch-wasm-bindgen": "watchlist src/wasi-rust -i src/wasi-rust/target -- yarn wasm-bindgen-dev",
```
Comment (happybeing) [inline package.json]: The command `watch-wasm-bindgen` now only regenerates the bindings whereas it used to compiles the Rust to wasm. It used to run `dev-wasm-bindgen` which ran `dev-wasm` (compile Rust to wasm) and then `wasm-bindgen` (generate bindings). Poor naming I know!

I like the way you've re-organised the naming of the script commands, so to fix this I think you can just change it to:
```
    "watch-build-dev": "watchlist src/wasi-rust -i src/wasi-rust/target -- yarn build-dev",
```
And change the README from `watch-wasm-bindgen` to `watch-build-dev`.

**Review by happybeing — COMMENTED** [review]
I like what you've done and suggested some changes. If you make changes and then push a new commit to main it should automatically update the pull request. I suggest you include the "small fix" too.

File: package.json (hunk 2 above)
Comment (kustomzone) [reply package.json]: Yup. Perfect!

File: README.md (hunk 1 above)
Comment (kustomzone) [reply README.md]: All done.

**Review by happybeing — CHANGES_REQUESTED** [review]
I see you've made a few changes to the README which are outside the scope of this PR. Rather than roll them into the set of changes for Windows support, please can you back out of those and create an issue so we can discuss your thinking. Thanks!

**Comment by happybeing** [comment]
Let me know if you want to update this or leave me to merge changes without the late README bits.

**Comment by happybeing** [comment]
I've merged commit 'e66a6e741111cdbd72bb8741550253cfd679e413' of https://github.com/kustomzone/p2p-git-portal-wasi into main and applied the changes I requested above, so closing this.

Thanks very much for this, it's good.

**Pull request closed without merging**